lib/repo.js

```diff
@@ -0,0 +1,52 @@
+import WBEdit from 'wikibase-edit'
+import { WBK } from 'wikibase-sdk'
+
+export default class WikibaseRepository {
+  constructor (origin, opts = {}) {
+    this.origin = origin
+
+    this.read = new WBK({
+      instance: origin
+    })
+
+    if (opts.oauth) {
+      this.edit = new WBEdit({
+        instance: origin,
+        credentials: {
+          oauth: opts.oauth
+        }
+      })
+    }
+  }
+
+  getContentLanguages () {
+    return fetch(`${this.origin}/w/api.php?action=query&meta=wbcontentlanguages&format=json`)
+      .then(r => r.json())
+      .then(body => Object.keys(body.query.wbcontentlanguages))
+  }
+
+  createEntities (...entities) {
+    if (!this.edit) {
+      return Promise.reject(new Error('Cannot edit a read only instance.'))
+    }
+    return Promise.all(entities.map(async entity => {
+      return this.edit.entity.create(entity)
```

Comment:
Does this stop the whole import if one entity fails to be created? I think I've convinced myself that we would rather have this continue even if one entity fails

Reply:
This rejects when the first promise in the given array rejects, so it's not entirely predictable how much work is done in such a case. We should be able to use `allSettled` though and get what you describe. Question though is: do we consider such a case a failure still and exit 1?

Reply:
Added this in https://github.com/wbstack/transferbot/pull/7/commits/9da79ce0b13fe806410fca476fcf5b1b7c858cee

Reply:
nice; part of my reasoning is that it means rerunning the job will be less painful in future (although the rerun will also error).